sdk/src/main/scala/com/horizen/account/api/http/AccountEthRpcRoute.scala

```diff
@@ -76,49 +75,37 @@ case class AccountEthRpcRoute(
    */
   def ethRpc: Route = post {
     withBasicAuth {
-      _ => {
-        entity(as[JsonNode]) { body =>
-
-          val json = {
-            // if the input json is an array a batch rpc request will be handled
-            // the single rpc request will retrieve from the input json and they will be processed by rpcHandler
-            // the position of the elements in the output will reflect their position in the input request
-            if(body.isArray()) {
-              if(!body.isEmpty) {
-                val responseList = mutable.MutableList[ApiResponse]()
-                val iterator: util.Iterator[JsonNode] = body.asInstanceOf[ArrayNode].elements()
-                while (iterator.hasNext) {
-                  try {
-                    val rpcRequest = new RpcRequest(iterator.next())
-                    responseList += rpcHandler.apply(rpcRequest)
-                  } catch {
-                    case _: RpcException => responseList += new RpcResponseError(new RpcId(), RpcError.fromCode(RpcCode.InvalidRequest, null))
-                  }
-                }
-                SerializationUtil.serialize(responseList.toList)
-              }
-              // if the input json is an empty array the output will be a single invalidRequest rpc response
-              else {
-                SerializationUtil.serialize(new RpcResponseError(new RpcId(), RpcError.fromCode(RpcCode.InvalidRequest, null)))
-              }
+      _ =>
+        {
+          entity(as[JsonNode]) { body =>
+            val requests = if (body.isArray && !body.isEmpty) {
+              // if the input json is an array a batch rpc request will be handled
+              // the single rpc request will retrieve from the input json and they will be processed by rpcHandler
+              // the position of the elements in the output will reflect their position in the input request
+              body.iterator().asScala.toArray
+            } else {
+              // if the input json is not an array a single rpc request will be handled
+              Array(body)
             }
-            // if the input json is not an array a single rpc request will be handled
-            else {
-              val response = {
-                try {
-                  rpcHandler.apply(new RpcRequest(body))
-                } catch {
-                  case _: RpcException => new RpcResponseError(new RpcId(), RpcError.fromCode(RpcCode.InvalidRequest, null))
-                }
-              }
-              SerializationUtil.serialize(response)
+
+            val responses = requests.map(json => Try.apply(new RpcRequest(json)).map(rpcHandler.apply) match {
+              case Success(value) => value
+              case Failure(exception: RpcException) => new RpcResponseError(new RpcId(), exception.error);
```

Comment:
why do we "new RpcId()" instead of getting the real `id` from the request?

Reply:
This error case here only happens when there was a problem parsing the `RpcRequest`, which is why the Id is not used. I didn't change the behavior here, but you are right this is a little incomplete. If the id is valid it should be used, even if something else in the request failed to parse.. :thinking:

Reply:
ok, we will keep it as an improvement for future.